lib/models/Location.js

```diff
@@ -0,0 +1,46 @@
+const nyplCore = require('../load_nypl_core')
+
+class Location {
+  constructor ({ holdingLocation, recapCustomerCode, m2CustomerCode }) {
+    this.nyplCoreLocation = nyplCore.sierraLocations()[holdingLocation]
+    this.m2CustomerCode = m2CustomerCode
+    this.recapCustomerCode = recapCustomerCode
+    this.requestable = this.nyplCoreLocation?.requestable
+  }
+
+  get deliveryLocation () {
+    // requestable is false (not undefined) only if we know a sierra holding
+    // location is not requestable.
+    if (this.requestable !== false) {
+      return this.deliveryLocationByHoldingLocation || this.deliveryLocationsByRecapCustomerCode || this.deliveryLocationsByM2CustomerCode
+    }
+  }
+
+  get deliveryLocationByHoldingLocation () {
+    if (this.nyplCoreLocation?.sierraDeliveryLocations?.length) {
+      // It's mapped, but the sierraDeliveryLocation entities only have `code` and `label`
+      // Do a second lookup to populate `deliveryLocationTypes`
+      return this.nyplCoreLocation.sierraDeliveryLocations.map((deliveryLocation) => {
+        deliveryLocation.deliveryLocationTypes = this.nyplCoreLocation.deliveryLocationTypes
+        return deliveryLocation
+      })
+    }
+  }
+
+  get deliveryLocationsByRecapCustomerCode () {
+    if (nyplCore.recapCustomerCodes()[this.recapCustomerCode]?.sierraDeliveryLocations) {
+      return nyplCore.recapCustomerCodes()[this.recapCustomerCode].sierraDeliveryLocations
+    }
+  }
+
+  get deliveryLocationsByM2CustomerCode () {
+    if (nyplCore.m2CustomerCodes()?.[this.m2CustomerCode]?.sierraDeliveryLocations) {
```

Comment:
Should these also be checking `this.nyplCoreLocation.deliverableToResolution` before considering returning any locations for the given resolution method?

Reply:
Good catch. That is what the code was doing before. Do you remember why we didn't simply rely on the presence of a customer code to use that resolution?